Terminal.Gui/Views/TableView/TableView.cs

```diff
@@ -340,17 +341,15 @@ public override void OnDrawContent (Rect contentArea)
 					var colStyle = Style.GetColumnStyleIfAny (current.Column);
 					var colName = table.ColumnNames [current.Column];
 
-					/*
-					if (!Style.ShowVerticalHeaderLines && current.X - 1 >= 0) {
-						AddRune (current.X - 1, yh, (Rune)Style.SeparatorSymbol);
+					if (!Style.ShowVerticalHeaderLines && current.X > 0) {
+						AddRune (current.X - 1, yh, (Rune)Style.HeaderSeparatorSymbol);
 					}
-					*/
 
 					Move (current.X, yh);
 
-					if (current.Width - colName.Length > 0 &&
-						Style.ShowHorizontalHeaderThroughline &&
-						(!Style.ShowHorizontalHeaderOverline || !Style.ShowHorizontalHeaderUnderline)) {
+					if (current.Width - colName.Length > 0
+						&& Style.ShowHorizontalHeaderThroughline
+						&& (!Style.ShowHorizontalHeaderOverline || !Style.ShowHorizontalHeaderUnderline)) {
```

Comment:
Micro-optimization opportunity (also not in scope for this PR):

By DeMorgan's theorem, `(!Style.ShowHorizontalHeaderOverline || !Style.ShowHorizontalHeaderUnderline)` is equivalent to `!(Style.ShowHorizontalHeaderOverline && Style.ShowHorizontalHeaderUnderline)`

Each short-circuits if Style.ShowHorizontalHeaderOverline is false, but an extra unary `!` is avoided with the `&&` form if it is true, as there's only one in the latter case.

This is probably also applied by the compiler for optimized builds, though.

But a thought about that:

In places where debug and release builds may have different compiled output, it's a good idea to try to make them be the same.

Reply:
Haha, I don't need no stinkin' truth tables.  I wasn't aware there was a difference in compiled output in this case.  In fact, I guess I sort of always assumed that if there was, that a disjunction (with the potential for a "short-circuit") would be better, since (potentially) having to look up the additional value in memory would be much more expensive than doing the negation.

Reply:
Haha well as stated it is a micro-optimization and, in cases like that, my opinion is that clarity of intent is much more important than shaving off one operator method call in a subset of cases.

In this particular instance, they are identical except for the single situation I called out, so it's truly a super-micro-optimization.

But, for future reference, a lot of commentary I make, unless I put it in an actual change request review, are purely for visibility of opportunities, concepts, etc for discussion or for consideration for future work.

If I see an actual _problem_ that I think should be addressed before merge, I'll be pretty clear about it and request a change in the review.

The only thing about the code in this PR that can close to that for me is that we strongly prefer switches (either expressions or statements as appropriate) for control flow logic, when possible, both for style and performance reasons.

But that's IMO not important enough, at this stage, to block a PR like this one. 😊

Reply:
Oh and thanks for working on this.

Do you need any assistance, with regard to https://github.com/gui-cs/Terminal.Gui/pull/2574#issuecomment-1915487800 or anything else for this?

Reply:
Thanks.  Any assistance would be appreciated, and as I won't have time to work on this for the next few days there shouldn't be any duplication of effort.

Reply:
Done, but only by dint of allowing users to make the strange choice of using all three lines.

Also, regarding switch-case, I don't use them unless there are at least 3 options; here there are a few 2-option if-else's and 2-option if-else-if's without a default else.

Of course that's not to say that some cleanup couldn't be done; this still retains much of the way the original version was structured.